### ISSUE TITLE
Follow target stops being followed when reparenting

### ISSUE THREAD

**Issue by TranquilMarmot** [ramokz/phantom-camera#408]
### Issue description

This wasn't an issue in `0.7.3` and only showed up after updating to `0.8`.

In my game, I have two main `PhantomCamera3D` instances. Let's call them "Player camera" and "Vehicle camera". I swap between them by swapping the `priority` of each between `0` and `100`.

When swapping from "player camera" to "vehicle camera", another thing that happens is that the parent of the node that "player camera" is following gets re-parented, using `reparent` to be a child of the node that "vehicle camera" is following.

When swapping _back_ to "player camera" from "vehicle camera", the node that "player camera" is following gets re-parented back to its original node.

This breaks the `follow_target` behavior so that the follow target is no longer being followed, despite its phantom camera being active and having the `follow_target` properly set.

As a workaround, re-setting `follow_target` fixes this:

```gdscript
phantom_camera.follow_target = null
phantom_camera.follow_target = target
```

This is probably from changes made in https://github.com/ramokz/phantom-camera/pull/381 (which is from a bug I also reported, sorry 🙈 )

I _think_ it's because when `reparent` is called, the node exits and then re-enters the tree.

### Steps to reproduce

See the attached reproduction project; here's the revelant code:

```gdscript
func _unhandled_input(event: InputEvent):
	if event.is_action_pressed(&"ui_accept"):
		if not node_b.driving:
			# reparent the visuals to be inside the "car"
			# this way, the character is "driving"
			visuals_original_parent = visuals.get_parent()
			visuals.reparent(node_b)
			visuals.global_position = node_b.global_position

			phantom_camera.priority = 0
			node_b.phantom_camera.priority = 100

			node_b.driving = true
		else:
			# reparent the character back to their original parent
			visuals.global_position = visuals_original_parent.global_position
			visuals.reparent(visuals_original_parent)
			visuals_original_parent = null

			# after this, following the visuals is broken :(
			phantom_camera.priority = 100
			node_b.phantom_camera.priority = 0

			node_b.driving = false

			# WORKAROUND - uncomment to "fix" bug
			# phantom_camera.follow_target = null
			# phantom_camera.follow_target = visuals
```

### Minimal reproduction project

https://github.com/TranquilMarmot/phantom-camera-retarget-bug

See `node_a.gd`

**Comment by ramokz** [comment]
Thanks for the report!

Should be a very simple fix.

Try adding `_should_follow_checker()` to the `set_is_active()` functions so it looks like:

```gdscript
func set_is_active(node: Node, value: bool) -> void:
	if node is PhantomCameraHost:
		_is_active = value
		if value:
			_should_follow_checker()
	else:
		printerr("PCams can only be set from the PhantomCameraHost")
```

**Comment by ramokz** [comment]
Resolved in #414
